src/main/java/org/radarbase/appserver/controller/RadarUserController.java

```diff
@@ -54,205 +55,205 @@
-                                  user.getSubjectId(),
-                                  null))
-                      .collect(Collectors.toList()));
-      return ResponseEntity.ok(usersFinal);
-    } else {
-      return ResponseEntity.ok(users);
+    @Authorized(
+            permission = AuthPermissions.UPDATE,
+            entity = AuthEntities.SUBJECT,
+            permissionOn = PermissionOn.SUBJECT)
+    @PutMapping(
+            "/"
+                    + PathsUtil.PROJECT_PATH
+                    + "/"
+                    + PathsUtil.PROJECT_ID_CONSTANT
+                    + "/"
+                    + PathsUtil.USER_PATH
+                    + "/"
+                    + PathsUtil.SUBJECT_ID_CONSTANT)
+    public ResponseEntity<FcmUserDto> updateUserInProject(
+            @Valid @RequestBody FcmUserDto userDto,
+            @Valid @PathVariable String subjectId,
+            @Valid @PathVariable String projectId) {
+        userDto.setSubjectId(subjectId);
+        userDto.setProjectId(projectId);
+        FcmUserDto user = this.userService.updateUser(userDto);
+        return ResponseEntity.ok(user);
     }
-  }
 
-  @Authorized(permission = AuthPermissions.READ, entity = AuthEntities.SUBJECT)
-  @GetMapping("/" + PathsUtil.USER_PATH + "/user")
-  public ResponseEntity<FcmUserDto> getRadarUserUsingId(
-      HttpServletRequest request, @PathParam("id") Long id) {
-    if (id == null) {
-      throw new InvalidUserDetailsException("The given id must not be null!");
+    @Authorized(permission = AuthPermissions.READ, entity = AuthEntities.SUBJECT)
+    @GetMapping("/" + PathsUtil.USER_PATH)
+    public ResponseEntity<FcmUsers> getAllRadarUsers(HttpServletRequest request) {
+        FcmUsers users = this.userService.getAllRadarUsers();
+        if (authorization != null) {
+            // Filter the users based on access.
+            FcmUsers usersFinal =
+                    new FcmUsers()
+                            .setUsers(
+                                    users.getUsers().stream()
+                                            .filter(
+                                                    user ->
+                                                            authorization.hasPermission(
+                                                                    (RadarToken) request.getAttribute(AuthAspect.TOKEN_KEY),
+                                                                    AuthPermissions.READ,
+                                                                    AuthEntities.SUBJECT,
+                                                                    PermissionOn.SUBJECT,
+                                                                    user.getProjectId(),
+                                                                    user.getSubjectId(),
+                                                                    null))
+                                            .collect(Collectors.toList()));
```

Comment:
can you revert the formatting changes because IMO that was more readable? We should harmonise our style guides. There are some in the config folder, are you using these?

Reply:
Ah yes will make the changes. I think I'm using the guides from another project.